plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -804,11 +859,20 @@ function OPDSBrowser:showDownloads(item)
             end
         end
     end
-    if (#type_buttons % 2 == 1) then -- we need even number of type buttons
-        table.insert(type_buttons, {text = ""})
+    -- handles adding download button(s)
+    if (download_buttons ~= {}) then
+        if (#download_buttons == 1) then
+            table.insert(buttons, download_buttons)
+            table.insert(buttons, {})  -- add separator for stream buttons
+        elseif (#download_buttons % 2 == 1) then
+            table.insert(download_buttons, {text = ""})
+        end
+    end
+    if (#stream_buttons % 2 == 1) then -- we need even number of type buttons
```

Comment:
Don't you always have pairs of the stream buttons?

Reply:
Now that I think about it, that's true. I'll remove that.